sky/task.py

```diff
@@ -422,10 +427,49 @@ def from_yaml_config(
             task.set_outputs(outputs=outputs,
                              estimated_size_gigabytes=estimated_size_gigabytes)
 
-        resources = config.pop('resources', None)
-        resources = sky.Resources.from_yaml_config(resources)
-
-        task.set_resources({resources})
+        resources_config = config.pop('resources', None)
+        if resources_config and resources_config.get('any_of') is not None:
+            resources_set = set()
```

Comment:
We should `raise ValueError` (or use the schema check) when there are additional arguments provided in the `resources` field, e.g.,
```
if len(resources_config) > 1:
  raise ValueError(...)
```